### PR TITLE
BUG: Fix tests on Windows

### PR DESCRIPTION
**Pull request by larsoner**
So far this just removes all `BAD_MSVC`. Let's see if I can fix it by local debugging in Windows...

### CODE REVIEW

File: OpenMEEGConfigure.h.in [hunk 1]
```diff
@@ -27,3 +27,4 @@ static const char version[] = "@OpenMEEG_VERSION@";
 #cmakedefine USE_OPENBLAS
 #cmakedefine USE_VECLIB
 #cmakedefine CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION "@CMAKE_VS_WINDOWS_TARGET_PLATFORM_VERSION@"
+#cmakedefine WIN32
```
Comment (larsoner) [inline OpenMEEGConfigure.h.in:30]: Locally at least none of the `#if WIN32` conditionals were actually working on Windows. I fixed this by adding this line, and then changing all `#if WIN32` to `#ifdef WIN32`. Let's see if CIs agree...

**Comment by larsoner** [comment]
Okay we're green, feel free to review/merge @papadop @agramfort

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
@papadop I let you merge if happy

**Comment by papadop** [comment]
I do not know where we lost the WIN32 define of the command line (because it used to work). But the adopted solution is fine.
Note that the functions in filenames.h can be probably remove in favor of using std::filesystem, which is now standard in C++17. But this is to be done in a separate MR.

**Comment by papadop** [comment]
Thank you very much.

**Comment by larsoner** [comment]
Awesome, as I was checking/fixing the filename stuff I thought to myself "there must be some existing library for this stuff by now". I'll look at your follow up today!

**Pull request merged**